analyzers/its/regression-test.ps1

```diff
@@ -349,7 +344,7 @@ function VerifyUnexpectedIssues($actualIssues, $expectedIssues){
 function VerifyExpectedIssues ($actualIssues, $expectedIssues){
     $expectedButNotRaisedIssues = @()
 
-    foreach ($expectedIssue in $expectedIssues){
+    foreach ($expectedIssue in $expectedIssues | Where-Object { $ruleId -eq "" -or $_.IssueId -eq $ruleId }){
```

Comment:
It would be great to extract the filtering to a method. Currently it is duplicated (310, 347)

Reply:
It's just a small fragment, giving it a good name produces a lot of code

Reply:
I also gave it a try to define `filter`, but it wasn't easy to understand how it works